File: app/src/main/java/com/bridou_n/beaconscanner/events/Events.java
package com.bridou_n.beaconscanner.events;

import org.altbeacon.beacon.Beacon;
import org.altbeacon.beacon.Region;

import java.util.Collection;

/**
 * Created by bridou_n on 05/10/2016.
 */

public class Events {
    public static class RangeBeacon {
        private Collection<Beacon> beacons;
        private Region region;

        public RangeBeacon(Collection<Beacon> beacons, Region region) {
            this.beacons = beacons;
            this.region = region;
        }

        public Collection<Beacon> getBeacons() {
            return beacons;
        }

        public void setBeacons(Collection<Beacon> beacons) {
            this.beacons = beacons;
        }

        public Region getRegion() {
            return region;
        }

        public void setRegion(Region region) {
            this.region = region;
        }
    }

    public static class BluetoothState {
        private int state;

        public BluetoothState(int state) {
            this.state = state;
        }

        public int getState() {
            return state;
        }

        public void setState(int state) {
            this.state = state;
        }
    }
}

File: app/src/main/java/com/bridou_n/beaconscanner/events/Events.kt
package com.bridou_n.beaconscanner.events

import android.bluetooth.BluetoothAdapter
import com.bridou_n.beaconscanner.features.beaconList.BeaconListActivity
import org.altbeacon.beacon.Beacon
import org.altbeacon.beacon.Region

/**
 * Created by bridou_n on 05/10/2016.
 */

class Events {
    class RangeBeacon(var beacons: Collection<Beacon>, var region: Region)

    class BluetoothState(var state: Int) {
        fun getBluetoothState() : BeaconListActivity.BluetoothState  {
            return when (state) {
                BluetoothAdapter.STATE_OFF -> BeaconListActivity.BluetoothState.STATE_OFF
                BluetoothAdapter.STATE_TURNING_OFF -> BeaconListActivity.BluetoothState.STATE_TURNING_OFF
                BluetoothAdapter.STATE_ON -> BeaconListActivity.BluetoothState.STATE_ON
                BluetoothAdapter.STATE_TURNING_ON -> BeaconListActivity.BluetoothState.STATE_TURNING_ON
                else -> BeaconListActivity.BluetoothState.STATE_OFF
            }
        }
    }
}
